plugins/zoom-to-fit/src/index.js

```diff
@@ -0,0 +1,230 @@
+    this.onZoomOutWrapper_ = Blockly.browserEvents.conditionalBind(
+        this.svgGroup_, 'mousedown', null, this.onClick_.bind(this));
+  }
+
+  /**
+   * Handle click event.
+   * @private
+   */
+  onClick_() {
+    this.workspace_.zoomToFit();
+    const uiEvent = new (Blockly.Events.get(Blockly.Events.CLICK))(
+        null, this.workspace_.id, 'zoom_reset_control');
+    Blockly.Events.fire(uiEvent);
+  }
+
+
+  /**
+   * Returns the bounding rectangle of the UI element in pixel units relative to
+   * the Blockly injection div.
+   * @return {!Blockly.utils.Rect} The plugin’s bounding box.
+   */
+  getBoundingRectangle() {
+    return new Blockly.utils.Rect(
+        this.top_, this.top_ + this.HEIGHT_,
+        this.left_, this.left_ + this.WIDTH_);
+  }
+
+  /**
+   * Positions the zoom-to-fit control.
+   * It is positioned in the opposite corner to the corner the
+   * categories/toolbox starts at.
+   * @param {!Blockly.MetricsManager.UiMetrics} metrics The workspace metrics.
+   * @param {!Array<Blockly.utils.Rect>} savedPositions List of rectangles that
+   *     are already on the workspace.
+   */
+  position(metrics, savedPositions) {
+    // Not yet initialized.
+    if (!this.verticalSpacing_) {
+      return;
+    }
+    if (metrics.toolboxMetrics.position === Blockly.TOOLBOX_AT_LEFT ||
+        (this.workspace_.horizontalLayout && !this.workspace_.RTL)) {
+      // Right corner placement.
+      this.left_ = metrics.viewMetrics.width + metrics.absoluteMetrics.left -
+          this.WIDTH_ - this.MARGIN_SIDE_ -
+          Blockly.Scrollbar.scrollbarThickness;
+    } else {
+      // Left corner placement.
+      this.left_ = this.MARGIN_SIDE_ + Blockly.Scrollbar.scrollbarThickness;
+    }
+
+    // Upper corner placement
+    const minTop = this.top_ =
+        metrics.absoluteMetrics.top + this.verticalSpacing_;
+    // Bottom corner placement
+    const maxTop = metrics.absoluteMetrics.top + metrics.viewMetrics.height -
+        this.HEIGHT_ - this.verticalSpacing_;
+    const placeBottom =
+        metrics.toolboxMetrics.position !== Blockly.TOOLBOX_AT_BOTTOM;
+    this.top_ = placeBottom ? maxTop : minTop;
+
+    // Check for collision and bump if needed.
+    let boundingRect = this.getBoundingRectangle();
+    for (let i = 0, otherEl; (otherEl = savedPositions[i]); i++) {
+      if (boundingRect.intersects(otherEl)) {
+        if (placeBottom) {
+          // Bump up
+          this.top_ = otherEl.top - this.HEIGHT_ - this.MARGIN_BOTTOM_;
+        } else {
+          this.top_ = otherEl.bottom + this.MARGIN_BOTTOM_;
+        }
+        // Recheck other savedPositions
+        boundingRect = this.getBoundingRectangle();
+        i = -1;
+      }
+    }
+    // Clamp top value within valid range.
+    this.top_ = Blockly.utils.math.clamp(minTop, this.top_, maxTop);
+
+    this.svgGroup_.setAttribute('transform',
+        'translate(' + this.left_ + ',' + this.top_ + ')');
+  }
+}
+
+// Blockly.Css.register([
```

Comment:
Do you need this?

Reply:
I didn't before, but now it's being used.